packages/account-v2/src/components/LeaveConfirm/LeaveConfirm.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import { useFormikContext } from 'formik';
+import { useHistory } from 'react-router-dom';
+import { Button, Modal, Text, useDevice } from '@deriv-com/ui';
+import UnsavedChanges from '../../assets/status-message/ic-unsaved-changes.svg';
+
+type TLeaveConfirm = {
+    onCancel?: () => void;
+    onLeave?: () => void;
+};
+
+export const LeaveConfirm = ({ onCancel, onLeave }: TLeaveConfirm) => {
+    const history = useHistory();
+    const { dirty } = useFormikContext();
+    const { isMobile } = useDevice();
+
+    const [showPrompt, setShowPrompt] = useState(false);
+    const [currentPath, setCurrentPath] = useState('');
+    const unblock = useRef<ReturnType<ReturnType<typeof useHistory>['block']>>();
+
+    useEffect(() => {
+        unblock.current = history.block((prompt: { pathname: React.SetStateAction<string> }) => {
+            if (dirty) {
+                setCurrentPath(prompt.pathname);
+                setShowPrompt(true);
+            }
+            return !dirty;
+        });
+        return () => {
+            unblock.current();
+        };
+    }, [dirty, history]);
+
+    const handleLeave = useCallback(() => {
```

Comment:
as I see Button component is not wrapped with memo(), so there is no sence to use useCallback